tests/unit_tests/lcls_tools/common/measurements/test_screen_beam_profile.py

```diff
@@ -0,0 +1,119 @@
+        self.test_instantiate_pydantic_objects()
+
+    def create_test_image(self, size: tuple, center: list, radius: int):
+        # make img that is a circle in the center of the image with known
+        # standard dev and mean. no imports, no calls to external or
+        # internal files.
+        image = np.zeros(size)
+        for y in range(image.shape[0]):
+            for x in range(image.shape[1]):
+                distance = np.sqrt((x - center[0]) ** 2 + (y - center[1]) ** 2)
+                if distance < radius:
+                    image[y, x] = 1
+        return image
+
+    def test_instantiate_pydantic_objects(self):
+        # creating image processing class
+        self.radius = 50
+        self.size = (800, 800)
+        self.center = [400, 400]
+        self.extent = [300, 300]
+        self.means = [150, 150]
+        self.sigmas = [30, 30]
+        self.amplitude = [99, 99]
+        self.offsets = [1, 1]
+        self.roi = ROI(center=[400, 400], extent=[300, 300])
+        self.image_processor = ImageProcessor(roi=self.roi)
+
+        self.pvs = {
+            "image": "ArrayData",
+            "n_bits": "N_OF_BITS",
+            "n_col": "Image:ArraySize1_RBV",
+            "n_row": "Image:ArraySize0_RBV",
+            "resolution": ":RESOLUTION",
+        }
+        self.metadata = {
+            "area": "TEST",
+            "beam_path": ["SC_TEST"],
+            "sum_l_meters": 99.99,
+        }
+        self.control_name = "OTRS:TEST:650:"
+
+        self.screen_pvs = ScreenPVSet(**self.pvs)
+        self.meta_data = Metadata(**self.metadata)
+        self.controls_information = ScreenControlInformation(
+            control_name=self.control_name, PVs=self.screen_pvs
+        )
+        self.screen_test = ScreenTest(
+            controls_information=self.controls_information, metadata=self.metadata
+        )
+        self.screen_test.image = self.create_test_image(
+            size=self.size, center=self.center, radius=self.radius
+        )
+        self.gauss_model = GaussianModel()
+        self.projection = ProjectionFit(model=self.gauss_model)
+
+        self.screen_beam_profile_measurement = ScreenBeamProfileMeasurement(
+            name=self.control_name,
+            device=self.screen_test,
+            image_processor=self.image_processor,
+            fitting_tool=self.projection,
+        )
+
+    def test_single_measure(self):
+        perform_single_measure = self.screen_beam_profile_measurement.single_measure()
+        self.assertIsInstance(perform_single_measure, dict)
+        assert len(perform_single_measure) == 2
+
+    def test_measure(self):
+        perform_measure = self.screen_beam_profile_measurement.measure()
+        self.assertIsInstance(perform_measure, list)
+        self.assertIsInstance(perform_measure[0], dict)
+        if isinstance(self.gauss_model, GaussianModel):
+            for key, val in perform_measure[0].items():
+                if key == "amplitude_x":
```

Comment:
Ideally, have these be exact values. Read and write the files to a file (e.g. json or yaml) and them diff that file with an expected output. You can use [`ndiff`] if the numbers are fuzzy. However, testing the fitting function itself is outside the scope of this module so inserting a blank image, for example, would yield a precise value after the fit (0). You would only need to test, say, mean_x and mean_y that they were both 0.